src/main/java/org/folio/entlinks/service/authority/AuthoritySourceFileService.java

```diff
@@ -182,6 +180,30 @@ private void validateOnCreate(AuthoritySourceFile entity) {
     entity.getAuthoritySourceFileCodes().forEach(this::validateSourceFileCode);
   }
 
+  private AuthoritySourceFile updateInner(UUID id, AuthoritySourceFile modified,
+                                          BiConsumer<AuthoritySourceFile, AuthoritySourceFile> publishConsumer) {
+    log.debug("update:: Attempting to update AuthoritySourceFile [id: {}]", id);
+
+    validateOnUpdate(id, modified);
+
+    var existingEntity = repository.findById(id).orElseThrow(() -> new AuthoritySourceFileNotFoundException(id));
+    var detachedExisting = new AuthoritySourceFile(existingEntity);
+    if (modified.getVersion() < existingEntity.getVersion()) {
+      throw OptimisticLockingException.optimisticLockingOnUpdate(
+        id, existingEntity.getVersion(), modified.getVersion());
+    }
+
+    updateSequenceStartNumber(existingEntity, modified);
+
+    copyModifiableFields(existingEntity, modified);
+
+    AuthoritySourceFile saved = repository.save(existingEntity);
```

Comment:
in order not to have the issue similar to the one we had in [MODELINKS-219](https://folio-org.atlassian.net/browse/MODELINKS-219) I suggest to use ```saveAndFlush``` or you can check if there is such issue by adding integration test similar to the one added in MODELINKS-219.

possible issue: this method runs inside a transaction and the ```saved``` result obtained by ```save``` method may not reflect all changes until the transaction is committed, but you are already using ```saved``` in ```publishConsumer.accept(saved, detachedExisting)``` to send the event and this happens before transaction commit.

Reply:
Thanks, Mukhiddin. It's done.